src/lib.rs

```diff
@@ -2224,72 +2437,19 @@ impl LightningNode {
             invalid_input("No on-chain funds to resolve")
         );
 
-        let lsp_fees =
-            self.calculate_lsp_fee(onchain_balance.msats.as_msats().to_amount_up(&None).sats)?;
-
-        let swap_info = self
-            .rt
-            .handle()
-            .block_on(self.sdk.receive_onchain(ReceiveOnchainRequest {
-                opening_fee_params: lsp_fees.lsp_fee_params,
-            }))
-            .ok();
-
-        let sat_per_vbyte = self.query_onchain_fee_rate()?;
-
-        let prepared_sweep = match self.prepare_sweep(
-            swap_info
-                .clone()
-                .map(|s| s.bitcoin_address)
-                .unwrap_or("1BitcoinEaterAddressDontSendf59kuE".to_string()),
-            sat_per_vbyte,
-        ) {
-            Ok(s) => s,
-            Err(RedeemOnchainError::InsufficientFunds { .. }) => return Ok(None),
-            Err(e) => runtime_error!(
-                RuntimeErrorCode::NodeUnavailable,
-                "Failed to prepare sweep: {e}"
-            ),
-        };
-
-        // Require onchain fees to be less than half of the onchain balance to leave some leeway
-        //  (for now, the onchain fee is just an estimation because the destination address is unknown)
-        if prepared_sweep.onchain_fee_amount.sats * 2 > onchain_balance.sats_round_down().sats {
-            return Ok(None);
-        }
-
-        let lsp_fees = self.calculate_lsp_fee(prepared_sweep.amount.sats)?;
+        let prepare_onchain_tx =
+            move |to_address: String, sat_per_vbyte: u32| -> Result<(Sats, Sats)> {
+                let x = self
```

Comment:
maybe `sweep_info` instead of `x`?

Reply:
yes, it flew under the radar. Fix [7c74716](https://github.com/getlipa/lipa-lightning-lib/pull/1215/commits/7c74716d858a1fbcd1493fa67c54cd66f1c62385)